### PR TITLE
Add ability to skip no streams access check

### PR DESCRIPTION
**Pull request by danotorrey**
## Description

Add ability to optionally skip no streams access check when re-using the `SearchPage`.

/nocl internal code change. 
## Motivation and Context

This can be useful when using the `SearchPage` component to render a search that targets OpenSearch Data Streams (which are not actual Stream entities). This check is only present to provide improved messaging to users who don't have any streams access, but that is not necessary in all cases. 

Also see https://github.com/Graylog2/graylog-plugin-enterprise/pull/8941 for additional motivation and context.

## How Has This Been Tested?

View any standard dashboard with a user who does not have stream access, and verify that below `No stream permissions` message is still displayed.

## Screenshots
image

### CODE REVIEW

File: graylog2-web-interface/src/views/pages/SearchPage.tsx [hunk 1]
```diff
@@ -56,14 +57,15 @@ const SearchPageTitle = ({ children }: { children: React.ReactNode }) => {
...
 const SearchPage = ({
-  children,
+  children = null,
   isNew,
   view: viewPromise,
   loadNewView: _loadNewView = defaultLoadNewView,
   loadView: _loadView = defaultLoadView,
   executionState: initialExecutionState = SearchExecutionState.empty(),
-  searchResult,
+  searchResult = null,
```
Comment (danotorrey) [inline graylog2-web-interface/src/views/pages/SearchPage.tsx:66]: I received this lint error once I modified this component. Is there any other better option other than defining a default null prop value?

**Review by kingzacko1 — APPROVED** [review]
Test successfully. Not sure about the frontend linter error, hopefully a frontend dev can help out!

**Review by zeeklop — APPROVED** [review]
Tested successfully. Thank you Dan. Nice catch

File: graylog2-web-interface/src/views/pages/SearchPage.tsx (hunk 1 above)
Comment (dennisoelkers) [reply graylog2-web-interface/src/views/pages/SearchPage.tsx:66]: While `null` is okay, the goto way to deal with this is assigning `undefined`. This seems to be redundant, but the linter rule is supposed to remind of props which are optional and the default argument is forgotten. The linter cannot decide for us if the intention is to make it `undefined` if the prop is not passed or some other value, so we want to be reminded that we should double-check what the default value should be. This was actually the same with `defaultProps` as well.

Comment (danotorrey) [reply graylog2-web-interface/src/views/pages/SearchPage.tsx:66]: Thanks for the feedback @dennisoelkers. Makes sense. @zeeklop has the same feedback for me that `undefined` is the preferred default, but he also though that null might work slightly better in this case. Let me know if you think we are good with `null` or if I should update those to `undefined`.

Comment (danotorrey) [reply graylog2-web-interface/src/views/pages/SearchPage.tsx]: I went ahead and moved those props over to use `undefined`.

**Pull request merged**